### PR TITLE
Miscellaneous small fixes

### PR DESCRIPTION
**Pull request by artivis**
Miscellaneous small fixes:

- group == operator constness
- tangent constexpr variables
- tl::optional missing return value
- examples: init random generator and reduce noise
- disable macos-11 in CI since the image is not ready yet (see https://github.com/actions/virtual-environments/issues/841)
- delete Travis config file since we're not using it anymore

### CODE REVIEW

File: examples/se3_sam.cpp [hunk 1]
```diff
@@ -349,7 +351,7 @@ int main()
 
     //// Simulator ###################################################################
     poses_simu. push_back(X_simu);
-    poses.      push_back(Xi + SE3Tangentd::Random());  // use very noisy priors
+    poses.      push_back(Xi + (SE3Tangentd::Random()*0.1));  // use very noisy priors
```
Comment (joansola) [inline examples/se3_sam.cpp:354]: remove the "very noise priors" comment if using `0.1*rand`

File: examples/se3_sam.cpp [hunk 2]
```diff
@@ -380,7 +382,7 @@ int main()
 
             // store
             poses_simu. push_back(X_simu);
-            poses.      push_back(Xi + SE3Tangentd::Random()); // use very noisy priors
+            poses.      push_back(Xi + (SE3Tangentd::Random()*0.1)); // use very noisy priors
```
Comment (joansola) [inline examples/se3_sam.cpp:385]: remove the "very noise priors" comment if using 0.1*rand

File: examples/se3_sam_selfcalib.cpp [hunk 3]
```diff
@@ -363,7 +365,7 @@ int main()
 
     //// Simulator ###################################################################
     poses_simu. push_back(X_simu);
-    poses.      push_back(Xi + SE3Tangentd::Random());  // use very noisy priors
+    poses.      push_back(Xi + SE3Tangentd::Random()*0.1);  // use very noisy priors
```
Comment (joansola) [inline examples/se3_sam_selfcalib.cpp:368]: remove the "very noise priors" comment if using 0.1*rand

File: examples/se3_sam_selfcalib.cpp [hunk 4]
```diff
@@ -395,7 +397,7 @@ int main()
 
             // store
             poses_simu. push_back(X_simu);
-            poses.      push_back(Xi + SE3Tangentd::Random()); // use very noisy priors
+            poses.      push_back(Xi + SE3Tangentd::Random()*0.1); // use very noisy priors
```
Comment (joansola) [inline examples/se3_sam_selfcalib.cpp:400]: remove the "very noise priors" comment if using 0.1*rand

**Review by joansola — COMMENTED** [review]
See some stupid remarks in the comments. Then I-ll approve.

File: examples/se3_sam.cpp (hunk 2 above)
Comment (artivis) [reply examples/se3_sam.cpp]: Fix : `// use noisy priors`.

File: examples/se3_sam.cpp (hunk 1 above)
Comment (artivis) [reply examples/se3_sam.cpp]: Fix : `// use noisy priors`.

File: examples/se3_sam_selfcalib.cpp (hunk 3 above)
Comment (artivis) [reply examples/se3_sam_selfcalib.cpp]: Fix : `// use noisy priors`.

File: examples/se3_sam_selfcalib.cpp (hunk 4 above)
Comment (artivis) [reply examples/se3_sam_selfcalib.cpp]: Fix : `// use noisy priors`.

Comment (joansola) [reply examples/se3_sam_selfcalib.cpp]: Yep sounds good 2 me

**Comment by artivis** [comment]
@joansola Can I get an approval to merge this PR as soon as CI is done?

**Review by joansola — APPROVED** [review]
All agreed

**Pull request merged**